src/action.c

```diff
@@ -547,40 +588,32 @@ actions_run(struct view *activator, struct server *server,
 			}
 			break;
 		case ACTION_TYPE_GO_TO_DESKTOP:
-			if (!arg) {
-				wlr_log(WLR_ERROR, "Missing argument for GoToDesktop");
-				break;
-			}
-			struct workspace *target;
-			const char *target_name = action_str_from_arg(arg);
-			target = workspaces_find(server->workspace_current, target_name);
-			if (target) {
-				workspaces_switch_to(target);
+			{
+				const char *target_name = action_str_from_arg(arg);
+				struct workspace *target =
+					workspaces_find(server->workspace_current, target_name);
+				if (target) {
+					workspaces_switch_to(target);
+				}
 			}
 			break;
 		case ACTION_TYPE_SEND_TO_DESKTOP:
-			if (view) {
-				const char *to = get_arg_value_str(action, "to", NULL);
-				if (!to) {
-					wlr_log(WLR_ERROR,
-						"Missing 'to' argument for SendToDesktop");
-					break;
-				}
-				bool follow = get_arg_value_bool(action, "follow", true);
-				struct workspace *target = workspaces_find(view->workspace, to);
-				if (target) {
-					view_move_to_workspace(view, target);
-					if (follow) {
-						workspaces_switch_to(target);
-					}
+			if (!view) {
+				break;
+			}
+			const char *to = get_arg_value_str(action, "to", NULL);
+			bool follow = get_arg_value_bool(action, "follow", true);
```

Comment:
Not sure if we should add an `assert(to)` here due to the default value of `get_arg_value_str()`. Its safe to do as the action would otherwise be deleted in the first place but we don't do it for the other required arguments either for the same reason.

Reply:
> Or maybe just add a comment here to say that its always available due to the config time check and potential removal of the action?

Yes, good idea. Agree.

Reply:
added: 455c6fc3095597adc34a3c47c42676ac31b4fdad